### ISSUE TITLE
Problem with Parallel Send Client From Echo Client

### ISSUE THREAD

**Issue by lexs2kari** [tgalal/yowsup#1934]
Hi guys,

I'm having a issue that I'm don't know what's going on. 
Here's my scenario:
- I have a django server with a REST method that uses the "SendClient" to send whatsapp messages;
- I have a python process (Echo Client) that's always running, listening the whatsapp incoming messages;

My Problem:
- When I start the "echo client" everything runs very well. But when the django service is invocked executing the REST method that send message using SendClient, the process with "echo client" stops from receiving message, it takes about 30 minutes to listen the messages again.

Anyone have any idea about what happens here?

1 - Sorry for my awful English.
2 - Sorry if it's something very simple that I'm not seeing.

**Comment by jlguardi** [comment]
You can't start two clients simultaneously with the same credentials. 
Create a new bot (from echoclient) wich starts the django thread and call a send method (not full stack object) to send out messages.

**Comment by lexs2kari** [comment]
Hi jlguardi

Thank you very much for your attention

I tried to do it, I started the django server from the layer of the EchoClient in a separated thread.
But I realized that the Django is a multi-process enviroment, so I can't share my layer instance to execute/call the "sendmessage" method through the Rest Service of Django.

I searched for a manner to share a unique instance of a object between the various process (like a Singleton in Java), but I didn't found anything integrated with django scenario.
